mobile/native/src/trade/order/handler.rs

```diff
@@ -1,46 +1,41 @@
+use crate::config;
 use crate::db;
 use crate::event;
 use crate::event::EventInternal;
-use crate::ln_dlc;
 use crate::trade::order::FailureReason;
 use crate::trade::order::Order;
 use crate::trade::order::OrderState;
-use crate::trade::position;
 use anyhow::bail;
 use anyhow::Context;
 use anyhow::Result;
-use std::time::Duration;
-use trade::ContractSymbol;
-use trade::TradeParams;
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
 use uuid::Uuid;
 
 pub async fn submit_order(order: Order) -> Result<()> {
     db::insert_order(order)?;
 
-    ui_update(order);
-
-    // TODO: remove this dummy
-    let dummy_trade_params = TradeParams {
-        pubkey: ln_dlc::get_node_info()?.pubkey,
-        // We set this to our pubkey as well for simplicity until we receive a match from the
-        // orderbook
-        pubkey_counterparty: ln_dlc::get_node_info()?.pubkey,
-        // We set this to our order as well for simplicity until we receive a match from the
-        // orderbook
-        order_id: order.id,
-        order_id_counterparty: "the orderbook will know".to_string(),
-        contract_symbol: ContractSymbol::BtcUsd,
-        leverage: order.leverage,
-        leverage_counterparty: 2.0,
-        quantity: order.quantity,
-        execution_price: 55_000.0,
-        // in 24h
-        expiry: Duration::from_secs(60 * 60 * 24),
-        oracle_pk: ln_dlc::get_oracle_pubkey()?,
+    // todo: this model should include the contract_symbol
+    let new_order = trade::NewOrder {
+        price: dec!(22_000), // todo: replace with actual btcusd rate price
+        quantity: Decimal::try_from(order.quantity)?,
+        maker_id: "".to_string(), // todo: what should be provided on the maker_id?
```

Comment:
This was just a dummy field. Imho this should be the sender's pk, e.g. if maker posts the limit order, then this is the maker's pk, if it is a market order by the trader, then this is the trader's pk.

Reply:
I see.. Then we should probably name it `trader_id` or something.